### PR TITLE
Port APIClient to use Http4s

### PR DESCRIPTION
**Pull request by nhanloukiala**
Addressing #224 

Changelog:
- **Http4s:** Http4s recommended using one shared client. The client is backed by a  `ContextShift` (functional wrapper of ExecutionContext), which handles yielding & switching thread pools behind the scene for users. 

- **Circe:** Json4s uses reflection to encode/decode(runtime). Circe uses macros to derive encoders/decoders (compile time). The former gives more freedom schema-wise and have better interop with Java classes. The latter is stricter type-wise, and therefore safer. 

Stuff to watch out for: 

- When calling `get` or `post`, always specify type parameter e.g. `get[String]`( or give the compiler a clear clue to infer correct type). This is due to HttpResponse decoded differently between `String`  vs. other types. [See](https://github.com/Constellation-Labs/constellation/pull/245/files#diff-eb93be29e89d6f3ad619d6f58ecd4c0eR80)
-  Import `io.circe.generic.auto._` when using the client, this instructs the macro to generates decoders when it compile. (Usually when missing it will throw missing implicit decoders).

Some open ideas (UPDATED): 
- ~~Replace `Json4s` with `Circe`. It's natively supported by Http4s and type-level. The encoder/decoder derivation is less boilerplate & more streamlined.~~ Too big to address in this PR, it can piggyback on the migration of server side when that happens. 
- ~~Keep vs. remove the blocking calls.~~ Same as above.
- ~~Use `IO` vs. `Future` monads across the apps. `IO` has more features and is more performant but `Future` is natively used in Akka.~~ The performance gain from IO is marginal and the coding style it enforces would probably confuses ppl more than helping.

### CODE REVIEW

**Comment by tylerprete** [comment]
@nhanloukiala really appreciate the contribution. I'll take a closer look later and we can discuss your ideas more

File: src/main/scala/org/constellation/primitives/TransactionExt.scala [hunk 1]
```diff
@@ -86,11 +86,6 @@ trait TransactionExt extends NodeData with Ledger with MetricsExt with PeerInfo
     tx
   }
 
-  def handleSendRequest(s: SendToAddress): StandardRoute = {
-    val tx = createTransaction(s.dst, s.amount, s.normalized)
-    complete(tx.prettyJson)
```
Comment (nhanloukiala) [inline src/main/scala/org/constellation/primitives/TransactionExt.scala]: `Accept: text/plain` is used for `complete(tx.prettyJson)`. Should be `complete(tx)` which uses `Accept: application/json`

**Comment by nhanloukiala** [comment]
Hi @tylerprete, that's no problem :). Hope the moving went smoothly. 

This week will be busy at my work as ppl coming back from vacation. I will update the PR bit by bit whenever I find time. ^^

**Comment by tylerprete** [comment]
@nhanloukiala I wanted to let you know we decided internally that this ticket is a high priority item for us -- so I'm going to go ahead and take it over. I'll use what you have in APIClientV2 as a base unless you have some objections. You could still work on the server-side, or we'd be happy to chat with you again about something else you could work on.

**Comment by nhanloukiala** [comment]
hi @tylerprete , sure please do. :) I had a great time working on that & glad it helps.

Been a bit chaotic on my side last week(end), but it's better now. Will be joining standup tomorrow :).

**Pull request closed without merging**